src/drawers/canvasDrawer.ts

```diff
@@ -20,6 +20,9 @@ export type CanvasDrawStep = (
  * This class is immutable (but has internal state) and shouldn't be extended.
  */
 export class CanvasDrawer implements IDrawer {
+  protected _lastDrawnData: any[];
+  protected _lastProjector: AttributeToAppliedProjector;
```

Comment:
Adding more state to canvasDrawer is a hack imo. Plottable has a big issue where our interface's responsibilities aren't well defined or respected, and this causes all sorts of broken behavior, illogical code, perf bugs, etc. We already have three representations of "drawn data": IPlotEntity, attrToProjector, and VisualPrimitive DOM elements. Consolidating this will give us 100x the return.

With that in mind, I think there are two things that we should do here:

1. The Plot calls `resolveAttributesSubsetWithStyles`, store that, and throw that into the Drawers - this keeps the Drawers stateless and puts the idea of a "last draw" only within Plot.

2. We refactor IDrawer's `getVisualPrimitives` to return our own `VisualPrimitive` interface (instead of a DOM Element), which has the bounds (again, preferably it has the resolved attrs directly so it's not rect specific), and use that everywhere.

Thoughts? This ask is obviously outside the scope of this PR but I want us to have a plan to fix this before we add more hack